src/components/FileTreeNode/index.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { TreefoldRenderProps } from 'react-treefold';
+import { ListGroup } from 'react-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import makeClassName from 'classnames';
+
+import { gettext } from '../../utils';
+import { TreeNode } from '../FileTree';
+import styles from './styles.module.scss';
+
+type PublicProps = TreefoldRenderProps<TreeNode>;
+
+class FileTreeNodeBase<TreeNodeType> extends React.Component<PublicProps> {
+  render() {
+    const {
+      node,
+      level,
+      isFolder,
+      isExpanded,
+      getToggleProps,
+      hasChildNodes,
+      renderChildNodes,
+    } = this.props;
+
+    let listGroupItemProps = {
+      className: makeClassName(styles.node, {
+        [styles.directoryNode]: isFolder,
+      }),
+    };
+
+    if (isFolder) {
+      listGroupItemProps = {
+        ...listGroupItemProps,
+        ...getToggleProps(),
```

Comment:
This doesn't seem to be covered by a test.

Reply:
Ah good catch, it's missing an assertion.